lib/logflare_web/controllers/api/source_controller.ex

```diff
@@ -3,9 +3,38 @@ defmodule LogflareWeb.Api.SourceController do
   alias Logflare.Sources
   alias Logflare.Users
 
+  action_fallback LogflareWeb.Api.FallbackController
+
   def index(%{assigns: %{user: user}} = conn, _) do
     user = Users.preload_sources(user)
     sources = Sources.preload_for_dashboard(user.sources)
     json(conn, sources)
   end
+
+  def create(%{assigns: %{user: user}} = conn, params) do
+    with params <- Map.put(params, "token", Ecto.UUID.generate()),
```

Comment:
we should shift down uuid generation to changeset level, actually. or even better, autogenerate in schema.

Reply:
will change it 👍